From lizf@cn.fujitsu.com  Fri Jun 13 17:33:08 2008
From: Li Zefan <lizf@cn.fujitsu.com>
Date: Fri, 13 Jun 2008 11:09:16 +0800
Subject: kobject: should use kobject_put() in kset-example
To: Greg KH <gregkh@suse.de>
Message-ID: <4851E4DC.6040603@cn.fujitsu.com>


We should call kobject_put() instead of kfree() if kobject_init_and_add()
returns an error, shouldn't we? Don't set up a bad example ;)

Signed-off-by: Li Zefan <lizf@cn.fujitsu.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 samples/kobject/kset-example.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/samples/kobject/kset-example.c
+++ b/samples/kobject/kset-example.c
@@ -211,7 +211,7 @@ static struct foo_obj *create_foo_obj(co
 	 */
 	retval = kobject_init_and_add(&foo->kobj, &foo_ktype, NULL, "%s", name);
 	if (retval) {
-		kfree(foo);
+		kobject_put(&foo->kobj);
 		return NULL;
 	}
 
